.travis.yml

```diff
@@ -1,20 +1,24 @@
 language: c
 
+# Default job
+os: linux
+sudo: required
+dist: trusty
+# Should spawn a linux-gcc job and a linux-clang job
+compiler: 
+  - gcc
+  - clang
+
+# Aditional testing on OSX with default compiler: should spawn one job
 matrix:
   include:
-    - os: linux
-      sudo: required
-      dist: trusty
-      group: edge
     - os: osx
+      env: COMPILER=clang
```

Comment:
It would be a matter of preference, but I would tolerate small amount of deviance if it helps other things like ease of development.

gcc-4 has a problem with long double. clang earlier than v3.9 has problems with openmp.
gcc-6 should work without any problem.
gcc-7.1 doesn't compile DFT correctly. We need to remove -ffast-math for gcc-7.1.